core/project_settings.cpp

```diff
@@ -629,19 +629,26 @@ Error ProjectSettings::_load_settings_text(const String &p_path) {
 }
 
 Error ProjectSettings::_load_settings_text_or_binary(const String &p_text_path, const String &p_bin_path) {
-	// Attempt first to load the text-based project.godot file
-	Error err_text = _load_settings_text(p_text_path);
-	if (err_text == OK) {
+	// Attempt first to load the binary project.godot file.
+	Error err = _load_settings_binary(p_bin_path);
+	if (err == OK) {
+		return OK;
+	} else if (err != ERR_FILE_NOT_FOUND) {
+		// If the file exists but can't be loaded, we want to know it.
+		ERR_PRINT("Couldn't load file '" + p_bin_path + "', error code " + itos(err) + ".");
+		return err;
+	}
+
+	// Fallback to text-based project.godot file if binary was not found.
+	err = _load_settings_text(p_text_path);
+	if (err == OK) {
 		return OK;
-	} else if (err_text != ERR_FILE_NOT_FOUND) {
-		// If the text-based file exists but can't be loaded, we want to know it
-		ERR_PRINT("Couldn't load file '" + p_text_path + "', error code " + itos(err_text) + ".");
-		return err_text;
+	} else if (err != ERR_FILE_NOT_FOUND) {
+		ERR_PRINT("Couldn't load file '" + p_text_path + "', error code " + itos(err) + ".");
+		return err;
 	}
 
-	// Fallback to binary project.binary file if text-based was not found
-	Error err_bin = _load_settings_binary(p_bin_path);
-	return err_bin;
+	return err;
```

Comment:
This logic is never executed as the logic above either succeed and returns `OK` or fails and returns an error.
I'll send a PR to fix the issue.

Reply:
It is executed if the error is different from `OK` (success) or `ERR_FILE_NOT_FOUND` (file doesn't exist, which is also a "success" so we don't abort and move to the next attempt).

What `err` are you getting?

Reply:
@akien-mga I created a separate PR (https://github.com/godotengine/godot/pull/47065) to fix the fallback.

> but that's not addressing the real issue.

There was no issue with the binary `project.godot` file. For the Android studio development use case, a regular Godot project (including the text-based `project.godot` file) is added to the `assets` directory of the Android project. And so the issue with the fallback was preventing that project from loading when the Android app was built.

I think the update in this PR overlooked the early return as it seems the intent was to log the unexpected error, not abort from it.

Reply:
The early return only happens if the error is not `ERR_FAIL_NOT_FOUND`. In your case it should be `ERR_FAIL_NOT_FOUND` if there's no file, so it shouldn't return.